### PR TITLE
Minor code fixes

### PR DESCRIPTION
**Pull request by builat**
* mutability removed from parseBytes method
* if block refactored in validate method.

### CODE REVIEW

**Comment by kushti** [comment]
@builat please take a look to ModifierValidator class, is it possible to reuse it, instead of building a Seq?

**Comment by builat** [comment]
@kushti well, if I understand Modifier Validator class, then definitely in possible to reuse it, but, anyway we should declare all the validation conditions mentioned in declared seq.

**Comment by catena2w** [comment]
@builat please use `ModifierValidator` and `ValidationResult` here, you may see an example here - https://github.com/ergoplatform/ergo/blob/master/src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala#L327.
P.S. `PoPoWProof` class is a bit outdated, so now it's style differs from other code, but it's better to make it similar to other modifiers validation.

**Comment by builat** [comment]
@catena2w not sure about ".toTry" method. But to be compliant with tests put it there.

File: src/main/scala/org/ergoplatform/modifiers/history/PoPoWProof.scala [hunk 1]
```diff
@@ -39,32 +41,39 @@ object PoPoWProof {
...
+      .validate(proof.m >= 1) {
+        error(s"m should positive, ${proof.m} given")
+      }
+      .validate(proof.suffix.head.interlinks(proof.i) == proof.innerchain.last.id) {
+        error(s"Incorrect link form suffix to innerchain in $proof")
+      }
+      .validate(proof.innerchain.length >= proof.m) {
+        error(s"Innerchain length is not enough in $proof")
+      }
+      .validate(!proof.innerchain.forall(h => powScheme.realDifficulty(h) < innerDifficulty)) {
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/modifiers/history/PoPoWProof.scala]: Seems like condition is not inversed here. Please check here and further.

**Pull request merged**